pkg/kv/kvserver/split/unweighted_finder.go

```diff
@@ -201,26 +201,31 @@ func (f *UnweightedFinder) NoSplitKeyCauseLogMsg() redact.RedactableString {
 		imbalanceAndTooManyContained)
 }
 
-// PopularKeyFrequency implements the LoadBasedSplitter interface.
-func (f *UnweightedFinder) PopularKeyFrequency() float64 {
+// PopularKey implements the LoadBasedSplitter interface.
+func (f *UnweightedFinder) PopularKey() PopularKey {
 	slices.SortFunc(f.samples[:], func(a, b sample) int {
 		return bytes.Compare(a.key, b.key)
 	})
 
 	currentKeyCount := 1
 	popularKeyCount := 1
+	var key roachpb.Key
 	for i := 1; i < len(f.samples); i++ {
 		if bytes.Equal(f.samples[i].key, f.samples[i-1].key) {
 			currentKeyCount++
 		} else {
 			currentKeyCount = 1
 		}
 		if popularKeyCount < currentKeyCount {
+			key = f.samples[i].key
 			popularKeyCount = currentKeyCount
 		}
 	}
 
-	return float64(popularKeyCount) / float64(splitKeySampleSize)
+	return PopularKey{
+		Key:       key,
```

Comment:
There's no data race in returning this byte slice up, correct? That is, the finder won't mutate this memory in the future?

Reply:
Hmmm..I'm not entirely sure. When you say mutate the memory, do you mean mutate the underlying slice? eg:

```
var k = roachpb.Key([]byte("test"))
finder.Record(k)
go func() { finder.PopularKey(k) }() 
go func() { k[0] = 'z' }() 
```

Reply:
The finder won't mutate the key bytes slice but it may swap out the key with another one in the next call. I think this is fine.